### ISSUE TITLE
bluetooth-battery@zamszowy: Need help to make it work in Mint 19.2/Cinnamon 4.2

### ISSUE THREAD

**Issue by Drugwash2** [linuxmint/cinnamon-spices-applets#4351]
```
 * Applet version/Build date: 1653224224 (**22.05.2022, 12:57:04**)
 * Cinnamon version: **4.2.4**
 * Distribution: **Mint 19.2**
 * Graphics hardware *and* driver used: Irrelevant
 * Snippet or attachment of ~/.cinnamon/glass.log: Irrelevant
 * Snippet or attachment of ~/.xsession-errors: Irelevant

 ```

**Notify author of applet**
@zamszowy

**Issue**
As is the applet doesn't even show an icon in panel. Using a try/catch managed to start it properly. However no device connection triggers a response. The only device recognized (somehow) is a type 1 `(dev.kind==1)` with 0% charge. System only recognizes AC power and Display device. The applet is needed for headphones only - no keyboard, mouse or other BT device here.

**Steps to reproduce**
Applet installed and enabled, no icon in panel.
Modified script to make it complete startup, then reloaded:
```
				try {
             	   if (dev.model || dev.serial)
                    // blacklist by default non mouse/kb/phone/gaming input/mediaplayer devices
                    this.blacklist_add(dev, true);
                } catch(e) {
                    // skip entirely devices without model and serial
                    continue;
                }

```
BT Disabled icon appears in panel, menu shows battery empty icon and 0%. Configuration only shows comment marks (#) that pile up after each applet restart.
Power up and connect any BT headphones or loudspeaker - no visible reaction from applet. Nothing appears in configuration, no setup dialog. No reaction on device power off either.

**Expected behaviour**
Dialog pops up on device connect, device added to the list or whatever the normal behavior would be. Icon would display battery level.

**Other information**
In order to approach the issue I have forcibly updated **UPower** to **0.99.11** from the default max 0.99.7 available for Mint 19.2. Running `upower --version` in Terminal yields:
```
UPower client version 0.99.11
UPower daemon version 0.99.11
```
Also **BlueZ** has been forcibly updated to **5.64**; first built from source and failed (wrong paths and/or configuration), then installed from Ubuntu 20.04 package. All related executables return 5.64 when queried for version in Terminal. Bluetooth connection works correctly with all headphones and loudspeaker. **Blueman v2.0.5** is used for managing connections.

In addition I have been running the latest available 5.4 kernel (**5.4.0-110-generic**) whereas I usually run **5.0.0-31-generic** as it seems to fit better my old hardware (Samsung NP-R580 notebook). Tried to install all the newer available in the 5.8, 5.11 and 5.13 line but they all failed to build the NVIDIA driver 340 so had to give up on those kernels.

Experimental features are enabled in the configuration file `/etc/bluetooth/main.conf` **General** section:
`Experimental = true`
Not sure if these features should have been enabled at compilation time or not, and if they have been in the Ubuntu package; my local build had all of them enabled but as mentioned the local installation failed due to unknown reasons.
Ran `systemctl daemon-reload` and `sudo systemctl restart bluetooth.service` after every change to the files.

Running `sudo /usr/lib/upower/upowerd --verbose &` in Terminal yields the following output:
<details>
  <summary>Click to expand log</summary>

```
dragos@Samsung-R580:~$ sudo /usr/lib/upower/upowerd --verbose &
[1] 3939
dragos@Samsung-R580:~$ TI:11:32:50	Acquired inhibitor lock (7)
TI:11:32:50	Starting upowerd version 0.99.11
TI:11:32:50	cannot find a keyboard backlight
TI:11:32:50	daemon now coldplug
TI:11:32:50	registering subsystem : power_supply
TI:11:32:50	taking a guess for power supply scope
TI:11:32:50	added native-path: ADP1
TI:11:32:50	object path = /org/freedesktop/UPower/devices/line_power_ADP1
TI:11:32:50	added ADP1
TI:11:32:50	emitting added: /org/freedesktop/UPower/devices/line_power_ADP1
TI:11:32:50	registering subsystem : usb
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-0:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-0:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-0:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.1/1-1.1:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.1/1-1.1:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.1/1-1.1:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.1/1-1.1:1.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.1/1-1.1:1.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.1/1-1.1:1.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.2/1-1.2.2:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.2/1-1.2.2:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.2/1-1.2.2:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.3
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.3
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.3
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.3/1-1.2.3:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.3/1-1.2.3:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.3/1-1.2.3:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4/1-1.2.4.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4/1-1.2.4.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4/1-1.2.4.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4/1-1.2.4.1/1-1.2.4.1:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4/1-1.2.4.1/1-1.2.4.1:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4/1-1.2.4.1/1-1.2.4.1:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4/1-1.2.4.2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4/1-1.2.4.2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4/1-1.2.4.2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4/1-1.2.4.2/1-1.2.4.2:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4/1-1.2.4.2/1-1.2.4.2:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4/1-1.2.4.2/1-1.2.4.2:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4/1-1.2.4:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4/1-1.2.4:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4/1-1.2.4:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-0:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-0:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-0:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.1/2-1.1:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.1/2-1.1:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.1/2-1.1:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.1/2-1.2.1:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.1/2-1.2.1:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.1/2-1.2.1:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.1/2-1.2.1:1.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.1/2-1.2.1:1.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.1/2-1.2.1:1.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.1/2-1.2.1:1.2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.1/2-1.2.1:1.2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.1/2-1.2.1:1.2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.1/2-1.2.3.1:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.1/2-1.2.3.1:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.1/2-1.2.3.1:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.2
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.2/2-1.2.3.2:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.2/2-1.2.3.2:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.2/2-1.2.3.2:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.4
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.4
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.4
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.4/2-1.2.3.4:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.4/2-1.2.3.4:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.4/2-1.2.3.4:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.4/2-1.2.3.4:1.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.4/2-1.2.3.4:1.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3.4/2-1.2.3.4:1.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.3/2-1.2.3:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.4
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.4
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.4
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.4/2-1.2.4:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.4/2-1.2.4:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.4/2-1.2.4:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.3
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.3
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.3
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.3/2-1.3:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.3/2-1.3:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.3/2-1.3:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.3/2-1.3:1.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.3/2-1.3:1.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.3/2-1.3:1.1
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.5
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.5
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.5
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.5/2-1.5:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.5/2-1.5:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.5/2-1.5:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1:1.0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1:1.0
TI:11:32:50	registering subsystem : usbmisc
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4/1-1.2.4.2/1-1.2.4.2:1.0/usbmisc/lp0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4/1-1.2.4.2/1-1.2.4.2:1.0/usbmisc/lp0
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.2/1-1.2.4/1-1.2.4.2/1-1.2.4.2:1.0/usbmisc/lp0
TI:11:32:50	registering subsystem : input
TI:11:32:50	not a switch [/sys/devices/LNXSYSTM:00/LNXPWRBN:00/input/input3/../capabilities/sw]
TI:11:32:50	invalid bitmask entry for /sys/devices/LNXSYSTM:00/LNXPWRBN:00/input/input3
TI:11:32:50	invalid bitmask entry for /sys/devices/LNXSYSTM:00/LNXPWRBN:00/input/input3/event3
TI:11:32:50	not a switch [/sys/devices/LNXSYSTM:00/LNXSYBUS:00/PNP0A08:00/device:00/LNXVIDEO:00/input/input6/../capabilities/sw]
TI:11:32:50	invalid bitmask entry for /sys/devices/LNXSYSTM:00/LNXSYBUS:00/PNP0A08:00/device:00/LNXVIDEO:00/input/input6
TI:11:32:50	invalid bitmask entry for /sys/devices/LNXSYSTM:00/LNXSYBUS:00/PNP0A08:00/device:00/LNXVIDEO:00/input/input6/event5
TI:11:32:50	not a switch [/sys/devices/LNXSYSTM:00/LNXSYBUS:00/PNP0C0C:00/input/input1/../capabilities/sw]
TI:11:32:50	invalid bitmask entry for /sys/devices/LNXSYSTM:00/LNXSYBUS:00/PNP0C0C:00/input/input1
TI:11:32:50	invalid bitmask entry for /sys/devices/LNXSYSTM:00/LNXSYBUS:00/PNP0C0C:00/input/input1/event1
TI:11:32:50	not a switch [/sys/devices/LNXSYSTM:00/LNXSYBUS:00/PNP0C0D:00/input/input0/../capabilities/sw]
TI:11:32:50	no device file: /sys/devices/LNXSYSTM:00/LNXSYBUS:00/PNP0C0D:00/input/input0
TI:11:32:50	watching /dev/input/event0 (9)
TI:11:32:50	using /sys/devices/LNXSYSTM:00/LNXSYBUS:00/PNP0C0D:00/input/input0/event0 for lid event
TI:11:32:50	lid_is_closed = no
TI:11:32:50	lid_is_present = yes
TI:11:32:50	added /sys/devices/LNXSYSTM:00/LNXSYBUS:00/PNP0C0D:00/input/input0/event0
TI:11:32:50	not a switch [/sys/devices/LNXSYSTM:00/LNXSYBUS:00/PNP0C0E:00/input/input2/../capabilities/sw]
TI:11:32:50	invalid bitmask entry for /sys/devices/LNXSYSTM:00/LNXSYBUS:00/PNP0C0E:00/input/input2
TI:11:32:50	invalid bitmask entry for /sys/devices/LNXSYSTM:00/LNXSYBUS:00/PNP0C0E:00/input/input2/event2
TI:11:32:50	not a switch [/sys/devices/pci0000:00/0000:00:01.0/0000:02:00.1/sound/card1/input11/../capabilities/sw]
TI:11:32:50	not a lid: /sys/devices/pci0000:00/0000:00:01.0/0000:02:00.1/sound/card1/input11
TI:11:32:50	not a lid: /sys/devices/pci0000:00/0000:00:01.0/0000:02:00.1/sound/card1/input11/event10
TI:11:32:50	not a switch [/sys/devices/pci0000:00/0000:00:01.0/0000:02:00.1/sound/card1/input12/../capabilities/sw]
TI:11:32:50	not a lid: /sys/devices/pci0000:00/0000:00:01.0/0000:02:00.1/sound/card1/input12
TI:11:32:50	not a lid: /sys/devices/pci0000:00/0000:00:01.0/0000:02:00.1/sound/card1/input12/event11
TI:11:32:50	not a switch [/sys/devices/pci0000:00/0000:00:01.0/0000:02:00.1/sound/card1/input13/../capabilities/sw]
TI:11:32:50	not a lid: /sys/devices/pci0000:00/0000:00:01.0/0000:02:00.1/sound/card1/input13
TI:11:32:50	not a lid: /sys/devices/pci0000:00/0000:00:01.0/0000:02:00.1/sound/card1/input13/event12
TI:11:32:50	not a switch [/sys/devices/pci0000:00/0000:00:01.0/0000:02:00.1/sound/card1/input14/../capabilities/sw]
TI:11:32:50	not a lid: /sys/devices/pci0000:00/0000:00:01.0/0000:02:00.1/sound/card1/input14
TI:11:32:50	not a lid: /sys/devices/pci0000:00/0000:00:01.0/0000:02:00.1/sound/card1/input14/event13
TI:11:32:50	not a switch [/sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.1/1-1.1:1.0/input/input15/../capabilities/sw]
TI:11:32:50	invalid bitmask entry for /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.1/1-1.1:1.0/input/input15
TI:11:32:50	invalid bitmask entry for /sys/devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.1/1-1.1:1.0/input/input15/event14
TI:11:32:50	not a switch [/sys/devices/pci0000:00/0000:00:1b.0/sound/card0/input10/../capabilities/sw]
TI:11:32:50	not a lid: /sys/devices/pci0000:00/0000:00:1b.0/sound/card0/input10
TI:11:32:50	not a lid: /sys/devices/pci0000:00/0000:00:1b.0/sound/card0/input10/event9
TI:11:32:50	not a switch [/sys/devices/pci0000:00/0000:00:1b.0/sound/card0/input9/../capabilities/sw]
TI:11:32:50	not a lid: /sys/devices/pci0000:00/0000:00:1b.0/sound/card0/input9
TI:11:32:50	not a lid: /sys/devices/pci0000:00/0000:00:1b.0/sound/card0/input9/event8
TI:11:32:50	not a switch [/sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.4/2-1.2.4:1.0/0003:04B3:310C.0001/input/input8/../capabilities/sw]
TI:11:32:50	invalid bitmask entry for /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.4/2-1.2.4:1.0/0003:04B3:310C.0001/input/input8
TI:11:32:50	invalid bitmask entry for /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.4/2-1.2.4:1.0/0003:04B3:310C.0001/input/input8/event7
TI:11:32:50	invalid bitmask entry for /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.4/2-1.2.4:1.0/0003:04B3:310C.0001/input/input8/mouse1
TI:11:32:50	not a switch [/sys/devices/platform/i8042/serio0/input/input4/../capabilities/sw]
TI:11:32:50	invalid bitmask entry for /sys/devices/platform/i8042/serio0/input/input4
TI:11:32:50	invalid bitmask entry for /sys/devices/platform/i8042/serio0/input/input4/event4
TI:11:32:50	not a switch [/sys/devices/platform/i8042/serio1/input/input7/../capabilities/sw]
TI:11:32:50	invalid bitmask entry for /sys/devices/platform/i8042/serio1/input/input7
TI:11:32:50	invalid bitmask entry for /sys/devices/platform/i8042/serio1/input/input7/event6
TI:11:32:50	invalid bitmask entry for /sys/devices/platform/i8042/serio1/input/input7/mouse0
TI:11:32:50	not a switch [/sys/devices/virtual/input/mice/../capabilities/sw]
TI:11:32:50	not a switch [/sys/devices/virtual/input/mice/capabilities/sw]
TI:11:32:50	registering subsystem : hid
TI:11:32:50	failed to coldplug /sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.2/2-1.2.4/2-1.2.4:1.0/0003:04B3:310C.0001
TI:11:32:50	on_battery = no
TI:11:32:50	warning_level = none
TI:11:32:50	daemon now not coldplug
TI:11:32:50	name lost, exiting
TI:11:32:50	Released inhibitor lock
```

</details>
When a device is connected the message **BlueZ appeared** is inserted right before the last message (Released inhibitor lock).
However, there may be something else wrong, since the other day there used to be messages when connecting/disconnecting devices, something about ignoring... can't remember what. That proved the system acknowledged the devices.

There also were files created in new folder in `/sys/devices/virtual/input` besides the static mice folder. There are no new folders and files now upon connecting devices.

It should be noted that the notebook has no battery at all, it has been physically pulled out since it was completely broken (burnt fuses, empty cells).

Any help would be appreciated. Since two of the headphones have a relatively long battery life - 20h and 40h respectively - it would be very useful to know when any of them needs recharging before starting a long audio session.

**Comment by zamszowy** [comment]
Hi @Drugwash2, thanks for the report. I admit that ignoring 'empty devices' logic is messed up right now and I'm in the middle of correcting this.
Nevertheless, after your fix, if upower recognized your headset, it should be displayed in the applet. Could you please paste the output of this command: `upower -d` before you connect the headset, and then after it's successfully connected?

**Comment by Drugwash2** [comment]
The output is virtually the same save for the differing timestamps. Tried that too these last few days, also tried `upower -e` which displays the shorter version.
<details>
  <summary>Click to reveal output</summary>
```
dragos@Samsung-R580:~$ upower -d
Device: /org/freedesktop/UPower/devices/line_power_ADP1
  native-path:          ADP1
  power supply:         yes
  updated:              Sun 22 May 2022 08:00:25 AM EEST (195571 seconds ago)
  has history:          no
  has statistics:       no
  line-power
    warning-level:       none
    online:              yes
    icon-name:          'ac-adapter-symbolic'

Device: /org/freedesktop/UPower/devices/DisplayDevice
  power supply:         no
  updated:              Sun 22 May 2022 08:00:26 AM EEST (195570 seconds ago)
  has history:          no
  has statistics:       no
  unknown
    warning-level:       none
    icon-name:          'battery-missing-symbolic'

Daemon:
  daemon-version:  0.99.11
  on-battery:      no
  lid-is-closed:   no
  lid-is-present:  yes
  critical-action: PowerOff
dragos@Samsung-R580:~$ upower -d
Device: /org/freedesktop/UPower/devices/line_power_ADP1
  native-path:          ADP1
  power supply:         yes
  updated:              Sun 22 May 2022 08:00:25 AM EEST (195602 seconds ago)
  has history:          no
  has statistics:       no
  line-power
    warning-level:       none
    online:              yes
    icon-name:          'ac-adapter-symbolic'

Device: /org/freedesktop/UPower/devices/DisplayDevice
  power supply:         no
  updated:              Sun 22 May 2022 08:00:26 AM EEST (195601 seconds ago)
  has history:          no
  has statistics:       no
  unknown
    warning-level:       none
    icon-name:          'battery-missing-symbolic'

Daemon:
  daemon-version:  0.99.11
  on-battery:      no
  lid-is-closed:   no
  lid-is-present:  yes
  critical-action: PowerOff
dragos@Samsung-R580:~$ 
```
</details>

**Comment by Drugwash2** [comment]
Well, that's about the problem I'm trying to fix. Been banging my head against the wall these days.
However, just now managed to get a folder with files created again upon connecting headphones, after resetting the daemon and the BT service. Some of the files/folders could not be captured.
_Is this platform really so stupid as I always thought? Can't attach a zip._ Trying a Dropbox link: [download](https://www.dropbox.com/s/9s4r9to9g93sg42/BTheadphones-input35.zip?dl=1)
The folder was in `/sys/devices/virtual/input/` as `input35`. Maybe some information there could shed some light.

There is no **Batteries** section in `System Settings -> Power Management` here (headphones were connected); it never was as far as I can recall:
![Screenshot from 2022-05-24 15-01-56](https://user-images.githubusercontent.com/29117024/170030075-45726d4f-c541-448d-b692-81ea135fac5c.png)

**Comment by zamszowy** [comment]
It could be that the BlueZ is not advertising battery info at all for your headset. What is bluetoothctl saying about it?
>bluetoothctl
>devices
>info [headset mac address here], eg: info 00:01:02:03:04:05

**Comment by Drugwash2** [comment]
Hm, this is possibly the start of a breakthrough. What I failed to mention from the very beginning is that the machine has two BT adapters: one is an internal Intel built into the Wi-Fi card and one is an external CSR dongle connected through a couple of daisy-chained USB hubs. Obviously I use the external one because it has a far greater power than the internal one, but I recall the CSR is not much fancied by the BT subsystem.

Now, after launching those commands I realized the BT subsystem obstinately clings onto the internal adapter **which is actually set as hidden in the GUI**, ignoring the other one. At least that is the logical outcome of the following output, knowing that the CSR has a much larger list of devices:
<details>
  <summary>Click to view output</summary>

```
dragos@Samsung-R580:~$ bluetoothctl
Agent registered
AdvertisementMonitor path registered
[bluetooth]# devices
Device 00:1A:7D:DA:71:11 CSR
Device 30:21:AE:B6:63:9C Street
Device 30:21:75:12:61:38 ABTS-S2
[bluetooth]# info 30:21:AE:B6:63:9C
Device 30:21:AE:B6:63:9C (public)
	Name: Street
	Alias: Street
	Class: 0x00240408
	Icon: audio-headset
	Paired: yes
	Bonded: yes
	Trusted: yes
	Blocked: no
	Connected: no
	LegacyPairing: no
	UUID: Headset                   (00001108-0000-1000-8000-00805f9b34fb)
	UUID: Audio Sink                (0000110b-0000-1000-8000-00805f9b34fb)
	UUID: A/V Remote Control Target (0000110c-0000-1000-8000-00805f9b34fb)
	UUID: A/V Remote Control        (0000110e-0000-1000-8000-00805f9b34fb)
	UUID: Handsfree                 (0000111e-0000-1000-8000-00805f9b34fb)
[bluetooth]# info 30:21:75:12:61:38
Device 30:21:75:12:61:38 (public)
	Name: ABTS-S2
	Alias: ABTS-S2
	Class: 0x00200418
	Icon: audio-headphones
	Paired: yes
	Bonded: yes
	Trusted: yes
	Blocked: no
	Connected: no
	LegacyPairing: no
	UUID: Headset                   (00001108-0000-1000-8000-00805f9b34fb)
	UUID: Audio Sink                (0000110b-0000-1000-8000-00805f9b34fb)
	UUID: A/V Remote Control Target (0000110c-0000-1000-8000-00805f9b34fb)
	UUID: A/V Remote Control        (0000110e-0000-1000-8000-00805f9b34fb)
	UUID: Handsfree                 (0000111e-0000-1000-8000-00805f9b34fb)
[bluetooth]#
```

</details>

In the list above **CSR** is the external adapter, **Street** is a pair of headphones (Krüger & Matz) and **ABTS-S2** is an Akai standalone clock+radio+bluetooth+thermometer (what I previously called a loudspeaker).

Now, how would one clearly and definitively lock out one of the adapters - more so being impossible to remove it physically - in order for the system to use only the preferred one? This may be the problem, or at least part of it.
_(There was a third BT adapter - a Broadcom - also internal, but that one had a mini-socket and could be physically disconnected as it was as lame as the Intel one; imagine what a mess would've been with three adapters)._

**Comment by zamszowy** [comment]
Please paste what `info MAC_ADDRESS_OF_DEVICE` inside bluetoothctl is showing when headset is connected.

**Comment by Drugwash2** [comment]
```
[ABTS-S2]# info 30:21:75:12:61:38
Device 30:21:75:12:61:38 (public)
	Name: ABTS-S2
	Alias: ABTS-S2
	Class: 0x00200418
	Icon: audio-headphones
	Paired: yes
	Bonded: yes
	Trusted: yes
	Blocked: no
	Connected: yes
	LegacyPairing: no
	UUID: Headset                   (00001108-0000-1000-8000-00805f9b34fb)
	UUID: Audio Sink                (0000110b-0000-1000-8000-00805f9b34fb)
	UUID: A/V Remote Control Target (0000110c-0000-1000-8000-00805f9b34fb)
	UUID: Advanced Audio Distribu.. (0000110d-0000-1000-8000-00805f9b34fb)
	UUID: A/V Remote Control        (0000110e-0000-1000-8000-00805f9b34fb)
	UUID: Handsfree                 (0000111e-0000-1000-8000-00805f9b34fb)
[ABTS-S2]# 
```

**Comment by zamszowy** [comment]
It still doesn't seem that BlueZ has battery information for those headsets (there should be "battery" line below all of those "UUID" lines).
Maybe you could enable just `BlueZ Experimental Bluetooth Quality Report` in `/etc/bluetooth/main.conf`, instead of all of them, to see if that changes anything, like this:
`Experimental = 330859bc-7506-492d-9370-9a6f0614037f`

**Comment by Drugwash2** [comment]
Had it that way before, no change. Tried it again now - again no change.
Maybe there is something wrong with UPower, some incompatibility at some level although it appears to work. When I fire up Power Statistics the Processor wakeups comes up with an error:
![Screenshot from 2022-05-26 00-16-30](https://user-images.githubusercontent.com/29117024/170368900-76c7248b-de7d-43e2-9ecf-5d5f96657e02.png)
The AC Adapter however reports correctly:
![Screenshot from 2022-05-26 00-18-35](https://user-images.githubusercontent.com/29117024/170369044-f3b912c6-6e87-47ba-9b66-f401313320d9.png)

**Comment by zamszowy** [comment]
I don't think it's Upower issue - of BlueZ doesn't have battery info, then Upower also won't get it.
Are you sure that your headset is sending battery info at all? You could check it by connecting them to some Android (10+ IIRC) device and see if it's reported there.

**Comment by Drugwash2** [comment]
Heck, I'm shooting in the dark. Unfortunately got no such device around. But thinking deeply, it's kinda slim chance that all my BT audio devices be defective/incomplete, more so when they've been bought new not earlier than two-three years ago. I mean we're talking about two Panasonic, two Krüger&Matz, an Akai and an LG.
![Screenshot from 2022-05-26 14-51-38](https://user-images.githubusercontent.com/29117024/170483153-ffab8180-7308-422a-9093-5c430e13fb6d.png)

Had too many troubles trying to make this work. After testing with the 5.4 kernel, returning to 5.0 made Cinnamon crash, so had to reinstall the NVIDIA video driver - for the second time in a few days. Incidentally, while looking through the logs I noticed at least one of the headphones was detected as a... keyboard! However, even so they should've been detected and displayed since all device types are enabled in the applet.
There must be something else here at play, maybe apparently unrelated. But no idea what that could be.

**Comment by zamszowy** [comment]
>[...] I noticed at least one of the headphones was detected as a... keyboard! However, even so they should've been detected and displayed since all device types are enabled in the applet.

Not really - only devices which report battery information would be displayed.

I'm sorry, but I'm out of ideas here of how to continue. :disappointed:
What I would focus on is trying to find out why there's no battery information reported in BlueZ - that seems to be the root cause.

**Comment by Drugwash2** [comment]
> Not really - only devices which report battery information would be displayed.

I mean, even if it was detected as a keyboard - or any other kind of BT device - it would've been picked up by the applet, if it reported a battery - which I presume BT keyboards do, at least some of them.

It's practically impossible that **none** of my devices reports battery. Obviously there's something deeper.

I got tired too, and because of this bug hunt I managed to broke Blueman so badly that not even the default/accepted version works correctly now - the devices list won't pop up, reporting Blueman is already running. Looks like I got my plate full.

Will put this issue on hold for a while until better ideas come to mind. Leaving this issue open in case someone else wants to chime in. Thank you very much for your help so far.

**Comment by Drugwash2** [comment]
Finally got to the bottom of this. By accident. The problem was the result of multiple factors:
- part of the BlueZ stack (namely `bluetoothd`) was running from a different location **and** without the experimental flag, due to an obscure configuration file
- Blueman was/is blocking the UPower detection signals - without providing the info itself either - so this applet didn't get notified of new devices
- most tests were performed using a device (Akai **ABTS-S2**) that turned out not to provide any battery info - or at least not the keys that the applet or Blueman are checking
- at some point I got very tired and lost focus; took a break only after about 36 hours of continuous fiddling with the system

So now that detection is working we can start fixing and improving. There are a few issues and a couple ideas that would need discussed. Unfortunately this platform is nowhere near usable to me so dunno how to proceed. Any ideas...?

**Comment by zamszowy** [comment]
@Drugwash2 Glad you were able to sort things out. I've prepared a change which should address the issue with empty device you were experiencing. 

Regarding other issues and ideas, I think that this github page would be a good place nevertheless, since it will track those issues/improvements for other users as well.

**Comment by Drugwash2** [comment]
Thanks, will check out the changes sometime later. There remains the issue of this applet and Blueman being somehow incompatible with each-other, at least in my setup, so will have to try and figure that out.

If I ever had any ideas they're long gone. Very bad memory. I still hate this platform, it's... impossible.

**Comment by Drugwash2** [comment]
OK, tested the applet with the new changes.

Not sure if popup notifications were spawned, I was in a full-screen application and turned off the headphones before minimizing it. There was no notification in the tray although headphones were beeping for critical battery status and critical threshold was set at 10%.

There was one time when connecting a different pair of headphones did not trigger an icon display. It was already excepted from the ignore list from a previous connection so it should've been detected and displayed. Could've been the Blueman clash or something else.

The idea of hiding the icon when nothing is connected doesn't seem too good in all situations. For one, when enabling the applet one doesn't have any visual confirmation of what it looks like or whether it actually was loaded; they'd have to open Looking Glass and check the messages. They'd also have to actually connect a BT device in order to be able to move the icon to a different place or panel. Then, changing any applet settings while no BT device is connected would require extra steps to reach its Settings window.
Therefore I'd lean towards making this optional and start with it disabled by default.
